gc.c

```diff
@@ -11020,6 +11022,19 @@ gc_info_decode(rb_objspace_t *objspace, const VALUE hash_or_key, const unsigned
       Qnil;
     SET(major_by, major_by);
 
+    if (orig_flags == 0) { /* set need_major_by only if flags not set explicitly */
```

Comment:
Since `need_major_flags` is only needed inside the `if` block, we can move it in.

```suggestion
    if (orig_flags == 0) { /* set need_major_by only if flags not set explicitly */
        unsigned int need_major_flags = objspace->rgengc.need_major_gc;

```